### ISSUE TITLE
`RedundantSuspendModifier` false-positives with extension functions

### ISSUE THREAD

**Issue by sschuberth** [detekt/detekt#7223]
## Expected Behavior

`RedundantSuspendModifier` should not complain for `suspend` functions that call `suspend` functions

## Observed Behavior

`RedundantSuspendModifier` is reported for this code:

    suspend fun deleteForOrganization(organizationId: Long, name: String) {
        db.dbQuery {
            infrastructureServiceRepository.deleteForOrganizationAndName(organizationId, name)
        }
    }

with

```
suspend fun <T> Database.dbQuery(
    transactionIsolation: Int = transactionManager.defaultIsolationLevel,
    readOnly: Boolean = transactionManager.defaultReadOnly,
    block: () -> T
): T =
    dbQueryCatching(transactionIsolation, readOnly, block).getOrThrow()
```

## Steps to Reproduce

Run Detekt with type resolution (`./gradlew detektMain`) on https://github.com/eclipse-apoapsis/ort-server/blob/main/services/infrastructure/src/main/kotlin/InfrastructureServiceService.kt

## Context

n/a

## Your Environment

* Version of detekt used: 1.23.6
* Version of Gradle used (if applicable): 8.7
* Gradle scan link (add `--scan` option when running the gradle task): n/a
* Operating System and version: Fedora Linus 39
* Link to your project (if it's a public repository): https://github.com/eclipse-apoapsis/ort-server

**Comment by atulgpt** [comment]
I have added TC for 

```kotlin
@Test
fun `does not report when suspend function is called in extension method`() {
    val code = """
        import kotlinx.coroutines.delay
        suspend fun foo() { delay(1000) }
        suspend fun String.bar() {
            foo()
        }
        suspend fun  String.baz() = foo()
    """.trimIndent()
    assertThat(subject.compileAndLintWithContext(env, code)).isEmpty()
}
```

And extension method detection working fine. Are you running with correct type resolution?

**Comment by sschuberth** [comment]
> Are you running detect tasks with the correct type resolution?

What do you mean by "correct"? But yes, I'm using type resolution in general, i.e. I run the `detektMain` task instead of the `detekt` task.

**Comment by atulgpt** [comment]
Hi @sschuberth it could detekt is unable to determine `dbQueryCatching(transactionIsolation, readOnly, block)` as suspend fun

**Comment by atulgpt** [comment]
Hi @sschuberth I tried running the project with `./gradlew detektMain` but it didn't report the `RedundantSuspendModifier` issue. 
Attaching SS
<img width="1635" alt="no-redundant-suspend-modifier-issue" src="https://github.com/detekt/detekt/assets/11026203/0cb86f54-c539-4f80-807a-5b09ddd3382f">

I am at 84d9a5c3debc9a6ea5dc493b1c36c2e871a0ba7a in main branch when I ran the `./gradlew detektMain` command

**Comment by sschuberth** [comment]
Thanks for checking, I also cannot reproduce the issue anymore.
